src/cryptographic_primitives/commitments/hash_commitment.rs

```diff
@@ -80,7 +80,7 @@ mod tests {
         let (_commitment, blind_factor) = HashCommitment::create_commitment(&message);
         let commitment2 =
             HashCommitment::create_commitment_with_user_defined_randomness(&message, &blind_factor);
-        assert_eq!(commitment2.to_str_radix(16).len(), SECURITY_BITS / 4);
+        assert!(commitment2.to_str_radix(16).len() / 2 <= SECURITY_BITS / 8);
```

Comment:
fyi, this is not fixing the issue. it basically. you now checking that the size of he commitment is larger than half the required size of bits. It means that the test will never fail (only with negligible probability) but if the hash output was indeed half the security bits length - this was not secure

Reply:
Now test checks that generated bytestring is not more than `SECURITY_BITS / 8` bytes length. Previously it used to check if generated string in hex representation is exactly `SECURITY_BITS / 4` symbols length which is not true with 1/16 probability. 

I didn't know that the test supposed to check security of random generator. Then it needs to collect ~100 samples and check whether they statistically secure